### DISCUSSION TITLE
is there a way to specify a static dev URL for AddNpmApp?

### DISCUSSION THREAD

**Discussion by sebastianwhiffen** [microsoft/aspire#1738] [Q&A] [answered]
I am using the following for my node frontend:

 `
builder.AddNpmApp("frontend", "../../app-frontend", builder.Environment.IsDevelopment() ? "dev" : "start")
    .WithReference(realtimerules_api)
    .WithServiceBinding(containerPort: 3000, scheme: "http", env: "PORT");
`

the current issue I am trying to solve is related to my authentication provider (Auth0) 

they require a static callback address for login/logout ect...

issue being that when AddNpmApp creates the port lifecycle it is seemingly random each time. and so I cannot specify the correct development callback address

is there a way to make this localhost address the same each time?

or any reason why this is not the default functionality?

Thank you in advance for your responses <3

**Comment by davidfowl** [comment, marked as answer]
> they require a static callback address for login/logout ect...

Specify a host port and pass that to the application.

```C#
builder.AddNpmApp("frontend", "../../app-frontend", builder.Environment.IsDevelopment() ? "dev" : "start")
.WithReference(realtimerules_api)
.WithServiceBinding(hostPort: 8085, containerPort: 3000, scheme: "http", env: "PORT");
```

Then you can hardcode 8085 in your auth configuration.

> or any reason why this is not the default functionality?

You can read about how networking works here https://learn.microsoft.com/en-us/dotnet/aspire/fundamentals/networking-overview

**Reply by stevedunn-marex** [reply]
Fast forward to 2025, and I can't seem to find the `WithServiceBinding` method...?

**Reply by davidfowl** [reply]
WithHttpEndpoint.

The linked networking doc also has this
